cmd/kola/kola.go

```diff
@@ -204,37 +208,52 @@ func writeProps() error {
 }
 
 func runList(cmd *cobra.Command, args []string) {
-	var w = tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0)
-	var testlist []item
-
+	var testlist []*item
 	for name, test := range register.Tests {
-		testlist = append(testlist, item{
+		item := &item{
 			name,
 			test.Platforms,
 			test.ExcludePlatforms,
-			test.Architectures})
+			test.Architectures,
+			test.Distros,
+			test.ExcludeDistros}
+		item.UpdateValues()
+		testlist = append(testlist, item)
 	}
 
 	sort.Slice(testlist, func(i, j int) bool {
 		return testlist[i].Name < testlist[j].Name
 	})
 
-	fmt.Fprintln(w, "Test Name\tPlatforms\tArchitectures")
-	fmt.Fprintln(w, "\t")
-	for _, item := range testlist {
-		fmt.Fprintf(w, "%v\n", item)
+	if !listJSON {
+		var w = tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0)
+
+		fmt.Fprintln(w, "Test Name\tPlatforms\tArchitectures\tDistributions")
+		fmt.Fprintln(w, "\t")
+		for _, item := range testlist {
+			fmt.Fprintf(w, "%v\n", item)
+		}
+		w.Flush()
+	} else {
+		out, err := json.Marshal(testlist)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "marshalling test list: %v", err)
+			os.Exit(1)
+		}
+		fmt.Fprintf(os.Stdout, string(out))
 	}
-	w.Flush()
 }
 
 type item struct {
 	Name             string
 	Platforms        []string
 	ExcludePlatforms []string
 	Architectures    []string
+	Distros          []string
+	ExcludeDistros   []string
```

Comment:
The `Exclude` fields should be `omitempty`.

Reply:
There's a subtle difference in `jq` between omit'd fields & fields containing empty lists (doing `.foo` if it's omit'd results in `null` vs `[]`). My thought was to leave it this way intentionally for easier scripting.

Reply:
Ah, I misread the code.  Do consumers need `Exclude*` at all?  Would it be sufficient to produce a list of platforms/distros which _are_ enabled for each test?

Reply:
Yeah, it should be sufficient to just provide an enabled list.